src/hooks/auth0-provider.js

```diff
@@ -0,0 +1,87 @@
+import React, {useReducer, useState} from 'react';
+import {useCallback, useMemo} from 'react';
+import jwt_decode from 'jwt-decode';
+import PropTypes from 'prop-types';
+import Auth0Context from './auth0-context';
+import Auth0 from '../auth0';
+import reducer from './reducer';
+import {idTokenNonProfileClaims} from '../jwt/utils';
+
+const initialState = {
+  isLoading: true,
+  user: null,
+  error: null,
+};
+
+/**
+ * @ignore
+ */
+const getIdTokenProfileClaims = idToken => {
+  const payload = jwt_decode(idToken);
+
+  const profileClaims = Object.keys(payload).reduce((profile, claim) => {
+    if (!idTokenNonProfileClaims.includes(claim)) {
```

Comment:
`idTokenNonProfileClaims` probably makes more sense as a `Set` if it's going to be accessed by value

Reply:
Yes, I agree with you. I couldn't see that types like `Set` were available in the JavaScript Runtime being used [according to this](https://reactnative.dev/docs/javascript-environment) but now that I have the sample converted to use this hook, I'll just fire it up and test it 👍🏻

Stand by..

Reply:
Yeah this appears to work fine, fixed in 21a5e87